docs/acn-internals.md

```diff
@@ -0,0 +1,166 @@
+
+The aim of this document is to describe at a high-level
+the main implementation of the Agent Communication Network (ACN).
+
+## Introduction
+
+The ACN protocol implements transmission control over the ACN.
+The message serialization is based on 
+<a href="https://developers.google.com/protocol-buffers" target="_blank">Protocol Buffers</a>.
+and the definition of the data structures involved is defined
+<a href="https://github.com/fetchai/agents-aea/blob/develop/libs/go/libp2p_node/acn/acn_message.proto" target="_blank">here</a>.
+
+## Messages
+
+### Agent Record
+
+An _Agent Record_ is a data structure containing information about an agent connected to the ACN.
+
+> TODO: list and explain fields types and purpose
```

Comment:
This should be described as you point out. We also have this data structure currently present in two places which needs to be addressed.